File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/flutterobject/NativeObjectMessageSink.java
package com.valentingrigorean.arcgis_maps_flutter.flutterobject;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.util.HashMap;

public class NativeObjectMessageSink implements NativeMessageSink {
    private final String objectId;
    private final NativeMessageSink messageSink;
    private final HashMap<String, Object> data = new HashMap<>(3);


    public NativeObjectMessageSink(String objectId, NativeMessageSink messageSink) {
        this.objectId = objectId;
        this.messageSink = messageSink;
    }

    @Override
    public void send(@NonNull String method, @Nullable Object args) {
        data.clear();
        data.put("objectId", objectId);
        data.put("method", method);
        data.put("arguments", args);
        messageSink.send("messageNativeObject", data);
    }
}

File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/flutterobject/NativeObjectMessageSink.kt
package com.valentingrigorean.arcgis_maps_flutter.flutterobject

class NativeObjectMessageSink(
    private val objectId: String?,
    private val messageSink: NativeMessageSink?
) : NativeMessageSink {
    private val data = HashMap<String, Any?>(3)
    override fun send(method: String, args: Any?) {
        data.clear()
        data["objectId"] = objectId
        data["method"] = method
        data["arguments"] = args
        messageSink!!.send("messageNativeObject", data)
    }
}